api/server/server.go

```diff
@@ -40,7 +40,8 @@ import (
 	"github.com/docker/docker/utils"
 )
 
-type ServerConfig struct {
+// APIServerConfig provides the configuration for the API server
+type APIServerConfig struct {
```

Comment:
I'm not sure about this change, as ``APIServerConfig`` repeat the ``api/server`` part. ``Config`` would be simpler :blush:. Below you end up with ``apiserver.APIServerConfig`` in ``daemon.go`` which would look way cleaner like ``apiserver.Config``.